core/src/bytecode/value/mod.rs

```diff
@@ -680,6 +531,120 @@ impl NickelValue {
         }
     }
 
+    /// Returns a mutable typed reference to the content of the value. Returns `None` if the
+    /// underlying value block has a reference count greater than one. See [Self::content_make_mut]
+    /// in this case.
+    pub fn content_mut(&mut self) -> Option<ValueContentRefMut<'_>> {
+        match self.tag() {
+            // Safety: if `self.tag()` is `Pointer`, then the content of self must be a valid
+            // non-null pointer to a value block.
+            ValueTag::Pointer => unsafe {
+                let as_ptr = NonNull::new_unchecked(self.data as *mut u8);
+                let header = ValueBlockRc::header_from_raw(as_ptr);
+
+                if header.ref_count() != 1 {
+                    return None;
+                }
+
+                // Safety:
+                //  - additionally, the lifetime of the return `ValueContentRef<'_>` is tied to
+                //    `&mut self`, so the former won't outlive the value block.
+                //  - we've checked above that `ref_count` is `1`, and we have a mutable borrow
+                //    over `self`, so there can't be other active mutable borrows to the block
+                Some(match header.tag {
+                    BodyTag::Number => ValueContentRefMut::Number(
+                        ValueBlockRc::decode_mut_from_raw_unchecked(as_ptr),
```

Comment:
Does it work to `let rc = ValueBlockRc::decode_mut_from_raw_unchecked(as_ptr)`, so you don't have to repeat the part? Or is there some subtlety that I'm missing?